src/main/java/dev/railroadide/railroad/java/JDKManager.java

```diff
@@ -0,0 +1,178 @@
+            } catch (IOException e) {
+                // fallback to raw path if normalization fails
+                uniqueJDKs.putIfAbsent(jdk.path(), jdk);
+            }
+        }
+
+        return new ArrayList<>(uniqueJDKs.values());
+    }
+
+
+    private static JavaVersion getJavaVersion(String javaHomeOrPath) {
+        String javaExecutable = javaHomeOrPath;
+        File f = new File(javaHomeOrPath, "bin/" + JAVA_EXECUTABLE);
+        if (f.exists() && f.canExecute()) {
+            javaExecutable = f.getAbsolutePath();
+        }
+
+        try {
+            Process p = new ProcessBuilder(javaExecutable, "-version")
+                .redirectErrorStream(true).start();
+            try (BufferedReader br = new BufferedReader(new InputStreamReader(p.getInputStream()))) {
+                String line;
+                Pattern pattern = Pattern.compile("\"(\\d+)\\.(\\d+).*\"");
+                while ((line = br.readLine()) != null) {
+                    Matcher matcher = pattern.matcher(line);
+                    if (matcher.find()) {
+                        int major = Integer.parseInt(matcher.group(1));
+                        int minor = Integer.parseInt(matcher.group(2));
+                        return new JavaVersion(major, minor);
+                    }
+                }
+            }
+        } catch (IOException e) {
+            // Ignore
+        }
+        return null;
+    }
+
+    private static JDK createJDKFromJavaExecutable(String javaExecutablePath) {
+        File javaExe = new File(javaExecutablePath);
+        if (!javaExe.exists()) return null;
+
+        // Get home directory: parent of bin
+        File binDir = javaExe.getParentFile();
+        if (binDir == null) return null;
+        File javaHomeDir = binDir.getParentFile();
+        if (javaHomeDir == null) return null;
+
+        String javaHome = javaHomeDir.getAbsolutePath();
+        JavaVersion version = getJavaVersion(javaHome);
+        if (version != null) {
+            String name = javaHomeDir.getName(); // use folder name
+            return new JDK(javaHome, name, version);
+        }
+        return null;
+    }
+
+    private static String findJavaOnPath() {
+        String pathEnv = System.getenv("PATH");
+        if (pathEnv == null) return null;
+        String[] paths = pathEnv.split(File.pathSeparator);
+        for (String p : paths) {
+            File javaFile = new File(p, JAVA_EXECUTABLE);
+            if (javaFile.exists() && javaFile.canExecute()) return javaFile.getAbsolutePath();
+        }
+        return null;
+    }
+
+    private static List<Path> getPossibleJDKPaths() {
+        List<String> paths = switch (OperatingSystem.CURRENT) {
+            case WINDOWS -> WIN_JDK_PATHS;
+            case MAC -> MAC_JDK_PATHS;
+            case LINUX -> LINUX_JDK_PATHS;
+            case UNKNOWN -> List.of();
+        };
+        return paths.stream().map(Paths::get).toList();
```

Comment:
`Paths.get` is an old method that isn't needed anymore.
```suggestion
        return paths.stream().map(Path::of).toList();
```